module2/homework/nwdNww/nwdNww.hpp

```diff
@@ -1,11 +1,29 @@
 #pragma once
+#include <algorithm>
+#include <cstdlib>
 
-int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWD(int lhs, int rhs)
+{
+    int higher = std::max(std::abs(lhs), std::abs(rhs));
+    int lower = std::min(std::abs(lhs), std::abs(rhs));
+    int remainder = 0;
+    while (lower)
+    {
+        remainder = higher % lower;
+        higher = lower;
+        lower = remainder;
+    }
+
+    return higher;
 }
 
-int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWW(int lhs, int rhs)
+{
+    int nwd = NWD(lhs, rhs);
+    if (nwd)
```

Comment:
Implicit casting to boolean. Comparing to 0 is much clearer
```suggestion
    if (nwd != 0)
```